source/common/filter/config_discovery_impl.h

```diff
@@ -409,11 +479,59 @@ class HttpFilterConfigProviderManagerImpl
                                  ProtobufTypes::MessagePtr&& default_config,
                                  bool last_filter_in_filter_chain,
                                  const std::string& filter_chain_type,
-                                 absl::string_view stat_prefix) override {
-    return std::make_unique<HttpDynamicFilterConfigProviderImpl>(
-        subscription, require_type_urls, factory_context, std::move(default_config),
-        last_filter_in_filter_chain, filter_chain_type, stat_prefix);
-  }
+                                 absl::string_view stat_prefix) override;
+};
+
+// TCP listener filter
+class TcpListenerFilterConfigProviderManagerImpl
```

Comment:
`TcpListenerFilterConfigProviderManagerImpl` and `UdpListenerFilterConfigProviderManagerImpl` are basically same, except the stat prefix and the implementation of `DynamicFilterConfigProviderImpl`. Can we add template class for them? using the stat prefix and the implementation of `DynamicFilterConfigProviderImpl` as template parameter.

Reply:
ACK

Yes, we can have TCP/UDP listener and HTTP filter all share the ListenerFilterConfigProviderManagerImpl as base class with more abstraction as in the latest PR.

However, after some experiments and discussions, we decided not put   StatPrefix as template parameter since it actually bring more problems than benefits as stated in an earlier comments.